src/frontend/src/common/components/library/data_subview/components/CreateNSTreeModal/components/SampleFiltering/index.tsx

```diff
@@ -21,6 +140,12 @@ const SAMPLE_FILTERING_TOOLTIP_TEXT = (
   </div>
 );
 
+// Styling for the InputDropdown child of general Dropdown component
+const InputDropdownProps = {
+  sdsStage: "userInput",
+  sdsStyle: "square",
+} as const;
```

Comment:
what is this type assertion?

Reply:
It's a ["const assertion"](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) -- it's so hard to find good documentation on it, I have no idea how I first learned about it, but basically it says that the thing preceding the `as const` should be read-only. It's weird how it looks like a type assertion, but it's effectively the opposite. Instead of a normal type assertion saying, "Trust me on this one, compiler", a const assertion is saying, "Yo compiler, enforce immutability and yell at me if I get out of line".

The reason I needed it here was that TS was complaining that my `InputDropdownProps` were no good when I tried to use them as props later because they could have been modified since their declaration and couldn't be guaranteed to conform to the props interface. The `as const` lets TS trust me because it knows I can't have any code changing it before passing it.